src/dmd/expressionsem.d

```diff
@@ -7811,6 +7811,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             }
             if (!exp.lwr || !exp.upr)
             {
+                if (auto ts = tp.next.isTypeStruct())
+                {
+                    auto s = search_function(ts.sym, Id.index);
+                    if (!s) s = search_function(ts.sym, Id.slice);
+                    if (s)
+                    {
+                        auto fd = s.isFuncDeclaration();
+                        if ((fd && !fd.getParameterList().length) || s.isTemplateDeclaration())
+                        {
+                            exp.error(
+                                "need upper and lower bound to slice pointer, perhaps you meant `(*%s)[]`",
```

Comment:
```suggestion
                                "upper and lower bounds are needed to slice a pointer, perhaps you meant `(*%s)[]`",
```

Reply:
Both this message and the one below that it is based on should use the same wording.

Reply:
Since we are explicitly checking for the presence of `opIndex` I think we could mention that in the error message to as clear as possible. E.g: "need upper and lower bound to slice pointer %s" and then an error supplemental: "pointer %s points to a struct type that defines an `opIndex`, perhaps you meant `(*%s)[]` ? This way there can be no confusion.

Reply:
Yes

Reply:
Done